src/locomoset/models/features.py

```diff
@@ -0,0 +1,29 @@
+"""
+Helper functions for extracting model features on a dataset.
+"""
+import torch
+from datasets import Dataset
+from transformers.modeling_utils import PreTrainedModel
+
+
+def get_features(
+    processed_data: Dataset, model_head: PreTrainedModel, batch_size: int = 4
+) -> torch.tensor:
+    """Takes preprocessed image data and calls a model with its classification head
+    removed to generate features.
+
+    Args:
+        processed_data: Preprocessed images compatible with model_head and including
+            'pixel_values' as a key.
+        model_head: A model with its classification head removed. Takes pixel_values as
+            input and outputs logits.
+        batch_size: No. of images passed to model_head at once.
+
+    Returns:
+        Extracted model features.
+    """
+    return processed_data.map(
+        lambda image: model_head(image["pixel_values"]),
```

Comment:
Can chat about this one when you're back but is it the use of map here you think is unconventional or the way we've wrapped this into the parent `get_features` function? I think it follows how I see HuggingFace datasets used in general, e.g. https://huggingface.co/docs/datasets/image_process .

Agree it may turn out we need to be able to control some more parameters for performance etc. here later, but I don't think that needs to be part of this PR.

Reply:
ah sorry, non-conventional for passing anything through the network - typically see this for preprocessing data (I guess you could think of feature extraction as this??). Agree not necessary to change in this PR but wanted to flag - should have been clearer, sorry!

Reply:
Ah yeah because this is inference rather than preprocessing basically, I see what you mean